src/source/Ice/IceAgent.c

```diff
@@ -2537,12 +2555,21 @@ STATUS handleStunPacket(PIceAgent pIceAgent, PBYTE pBuffer, UINT32 bufferLen, PS
             break;
 
         default:
-            CHK_STATUS(hexEncode(pBuffer, bufferLen, NULL, &hexStrLen));
-            hexStr = MEMCALLOC(1, hexStrLen * SIZEOF(CHAR));
-            CHK(hexStr != NULL, STATUS_NOT_ENOUGH_MEMORY);
-            CHK_STATUS(hexEncode(pBuffer, bufferLen, hexStr, &hexStrLen));
-            DLOGW("Dropping unrecognized STUN packet. Packet type: 0x%02x. Packet content: \n\t%s", stunPacketType, hexStr);
-            SAFE_MEMFREE(hexStr);
+            if (!IS_STUN_PACKET(pBuffer)) {
+                CHK_STATUS(hexEncode(pBuffer, bufferLen, NULL, &hexStrLen));
+                hexStr = MEMCALLOC(1, hexStrLen * SIZEOF(CHAR));
+                CHK(hexStr != NULL, STATUS_NOT_ENOUGH_MEMORY);
+                CHK_STATUS(hexEncode(pBuffer, bufferLen, hexStr, &hexStrLen));
+                DLOGW("Dropping unrecognized STUN packet. Packet type: 0x%02x. Packet content: \n\t%s", stunPacketType, hexStr);
+                SAFE_MEMFREE(hexStr);
+            } else if (STUN_PACKET_IS_TYPE_ERROR(pBuffer)) {
+                CHK_STATUS(hexEncode(pBuffer, bufferLen, NULL, &hexStrLen));
+                hexStr = MEMCALLOC(1, hexStrLen * SIZEOF(CHAR));
+                CHK(hexStr != NULL, STATUS_NOT_ENOUGH_MEMORY);
+                CHK_STATUS(hexEncode(pBuffer, bufferLen, hexStr, &hexStrLen));
+                DLOGW("Error STUN packet. Packet type: 0x%02x. Packet content: \n\t%s", stunPacketType, hexStr);
```

Comment:
Only diff is the log line?  Then let's move the rest outside the if/elseif?

Reply:
No because this would needlessly confuse people seeing this warning. We'll print it if it's an error or if it's not something we recognize as STUN. Anything else we have handled above as needed.